### PR TITLE
fix: resolve 'ERR' and 'undefined' metrics and bids data

### PR DESCRIPTION
**Pull request by domw30**
[Associated Notion Card](https://www.notion.so/zerotech/4985d8ec0c2f4eb4860678eec483e6f6?v=2de07237786c4d62830797da093e6a04&p=1c9f32ec479541aa83825e12acf5882b&pm=s)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
- bug
- refactor

## 3. What is the old behaviour?
Undefined data displaying incorrectly
Screenshot 2022-11-23 at 12 11 48
Screenshot 2022-11-23 at 12 11 57

## 4. What is the new behaviour?
Undefined data handled

Screenshot 2022-11-23 at 12 10 59
Screenshot 2022-11-23 at 12 12 50

### CODE REVIEW

File: src/features/view-nft/Actions/Actions.tsx [hunk 1]
```diff
@@ -1,117 +1,150 @@
...
 			isVisible: isUserBid,
-			dataTestId: DataTestId.USER_BID,
-			buttonComponent: <CancelBidButton />,
 		},
-	};
+	];
 
-	const orderedActions = getOrderedActions(isOwnedByUser, actions);
-	const visibleActions = getVisibleActions(orderedActions);
+	const orderedActions = isOwnedByUser
```
Comment (domw30) [inline src/features/view-nft/Actions/Actions.tsx:76]: I think some room for improvement here

File: src/lib/util/bids/bids.ts [hunk 2]
```diff
@@ -13,22 +13,30 @@ export const sortBidsByTime = (bids: Bid[]) => {
...
+ * Returns sorted bids by amount and highest bid.
  * @param bids list of bids to sort
  */
 export const sortBidsByAmount = (bids: Bid[]) => {
-	return bids?.sort((a, b) =>
+	const sortedBids = bids?.sort((a, b) =>
 		BigNumber.from(a.amount).gte(BigNumber.from(b.amount)) ? -1 : 1,
 	);
+
+	const highestBid = sortedBids?.length >= 0 ? sortedBids[0] : undefined;
```
Comment (domw30) [inline src/lib/util/bids/bids.ts:24]: change condition operators from `>=` to `>`

File: src/lib/util/bids/bids.ts [hunk 3]
```diff
@@ -13,22 +13,30 @@ export const sortBidsByTime = (bids: Bid[]) => {
...
- * @param bids list of bids filter
+ * @param bids list of bids to filter
  */
 export const getUserBids = (accountId: string, bids: Bid[]) => {
 	const userBids = bids?.filter(
 		(b) => b.bidder.toLowerCase() === accountId?.toLowerCase(),
 	);
-	return sortBidsByAmount(userBids);
+
+	const highestUserBid = userBids?.length >= 0 ? userBids[0] : undefined;
```
Comment (domw30) [inline src/lib/util/bids/bids.ts:39]: change condition operators from `>=` to `>`

File: src/features/accept-bid/AcceptBidButton/AcceptBidButton.tsx [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 
-import { AcceptBidModal, AcceptBidModalProps } from '..';
+import { Bid } from '@zero-tech/zauction-sdk';
+
+import { AcceptBidModal } from '..';
 
 export interface AcceptBidButtonProps {
-	zna: AcceptBidModalProps['zna'];
-	bid: AcceptBidModalProps['bid'];
+	zna: string;
```
Comment (domw30) [inline src/features/accept-bid/AcceptBidButton/AcceptBidButton.tsx:8]: The `type` changes within this PR were previously discussed with Brett. If creating a sub component, then the `types` should be from the parent component. These were previously typed from the child component (wrong way round)

File: src/features/view-nft/NFTMetrics/NFTMetrics.tsx [hunk 5]
```diff
@@ -17,21 +16,32 @@ export const NFTMetrics: FC<NFTMetricsProps> = ({ zna }) => {
...
+
+	// let numberOfBids, lastSale, volumeString;
+
+	// if (!isLoadingBids && bids) {
+	// 	numberOfBids = formatNumber(bids.length || 0).toLocaleString();
+	// }
+
+	// if (!isLoadingMetrics && metrics) {
+	// 	lastSale = formatEthers(metrics.lastSale);
+	// 	volumeString = formatEthers(metrics.volume.all);
```
Comment (padala) [inline src/features/view-nft/NFTMetrics/NFTMetrics.tsx:41]: Can we remove this commented code?

**Review by padala — DISMISSED** [review]

**Pull request merged**